packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -103,35 +106,79 @@ function patch(n1: VNode, n2: VNode) {
             break;
 
         case VNodeType.Element:
-            patchElement(n1 as VElement, n2);
+            patchElement(n1 as VElement, n2, vm);
+            break;
+
+        case VNodeType.Slot:
+            patchSlot(n1 as VSlot, n2, vm);
             break;
 
         case VNodeType.CustomElement:
-            patchCustomElement(n1 as VCustomElement, n2);
+            patchCustomElement(n1 as VCustomElement, n2, vm);
             break;
     }
 }
 
-function mount(node: VNode, parent: ParentNode, anchor: Node | null) {
+function mount(node: VNode, parent: ParentNode, anchor: Node | null, vm: VM) {
     switch (node.type) {
         case VNodeType.Text:
-            mountText(node, parent, anchor);
+            mountText(node, parent, anchor, vm);
             break;
 
         case VNodeType.Comment:
-            mountComment(node, parent, anchor);
+            mountComment(node, parent, anchor, vm);
             break;
 
         case VNodeType.Element:
-            mountElement(node, parent, anchor);
+            mountElement(node, parent, anchor, vm);
             break;
 
         case VNodeType.CustomElement:
-            mountCustomElement(node, parent, anchor);
+            mountCustomElement(node, parent, anchor, vm);
+            break;
+
+        case VNodeType.Slot:
+            mountSlot(node, parent, anchor, vm);
             break;
     }
 }
 
+function mountSlot(vnode: VSlot, parent: ParentNode, anchor: Node | null, vm: VM) {
+    const {
+        sel,
+        data: { svg },
+        aChildren,
+    } = vnode;
+
+    // weather to render the slotted content or the default slot content.
+    const children: VNodes = aChildren ? aChildren.vnodes : vnode.children;
+
+    // default content is part of vm.
+    const slottedContentOwnerVM = aChildren ? aChildren.owner : vm;
+
+    if (vm.renderMode === RenderMode.Light) {
+        // we need to render the children as part of parentNode, which anchor should we use?
+        // @todo: we need to add an empty text at the beginning of children
+        mountVNodes(children, parent, null, 0, children.length, slottedContentOwnerVM);
+        return;
+    }
+
+    const namespace = isTrue(svg) ? SVG_NAMESPACE : undefined;
+    const elm = createElement(sel, namespace);
+    linkNodeToShadow(elm, vm); // the slot element is part of vm
+
+    // let's render the slot as an element.
+    // @todo: alternatively vnode as unknown as VElement
+    fallbackElmHook(elm, { ...vnode, type: VNodeType.Element }, vm); // the slot element is part of vm
```

Comment:
this will also work, and should be "faster" (don't know by how much), though it may compromise the typings. thoughts?

```suggestion
    fallbackElmHook(elm, vnode as unknown as VElement, vm); // the slot element is part of vm
```

Reply:
I agree it's faster because we're not unnecessarily creating an extra object. Let's not sacrifice perf for the benefit of typesafety. :)